dora/testcontainers-tests/src/test/java/alluxio/membership/MembershipManagerTest.java

```diff
@@ -0,0 +1,293 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.membership;
+
+import alluxio.membership.MembershipType;
+import alluxio.conf.Configuration;
+import alluxio.conf.PropertyKey;
+import alluxio.membership.AlluxioEtcdClient;
+import alluxio.membership.EtcdMembershipManager;
+import alluxio.membership.MembershipManager;
+import alluxio.membership.StaticMembershipManager;
+import alluxio.network.TieredIdentityFactory;
+import alluxio.util.CommonUtils;
+import alluxio.util.WaitForOptions;
+import alluxio.wire.TieredIdentity;
+import alluxio.wire.WorkerInfo;
+import alluxio.wire.WorkerNetAddress;
+import eu.rekawek.toxiproxy.model.ToxicDirection;
+import org.apache.log4j.PropertyConfigurator;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.ToxiproxyContainer;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+public class MembershipManagerTest {
+  private static final Network network = Network.newNetwork();
+  private static final int ETCD_PORT = 2379;
+  @Rule
+  public TemporaryFolder mFolder = new TemporaryFolder();
+
+  private static ToxiproxyContainer.ContainerProxy etcdProxy;
+
+  //Add for logging for debugging purpose
+  @BeforeClass
+  public static void init() {
+    PropertyConfigurator.configure("/Users/lucyge/Documents/github/alluxio/conf/log4j.properties");
```

Comment:
shall we use a rand path from tempfolder? any issue with a hardcoded path here?

Reply:
sorry this is a local change checked in, will remove it as part of final readiness